src/main/plugins/org.dita.base/xsl/preprocess/conrefImpl.xsl

```diff
@@ -487,102 +501,50 @@ See the accompanying LICENSE file for applicable license.
     <xsl:param name="conref-filename" tunnel="yes" as="xs:string?"/>
-    <xsl:attribute name="id">
-      <xsl:choose>
-        <xsl:when test="exists($conref-filename)">
-          <xsl:value-of select="generate-id(..)"/>
-        </xsl:when>
-        <xsl:otherwise>
-          <xsl:value-of select="."/>
-        </xsl:otherwise>
-      </xsl:choose>
-    </xsl:attribute>
-  </xsl:template>
-
-  <xsl:template name="generate-href">
-    <xsl:param name="current-relative-path" tunnel="yes" as="xs:string"/>
-    <xsl:param name="conref-filename" tunnel="yes" as="xs:string"/>
-    <xsl:param name="topicid" tunnel="yes" as="xs:string?"/>
-    <xsl:param name="elemid" tunnel="yes" as="xs:string?"/>
-    <xsl:param name="conref-source-topicid" tunnel="yes" as="xs:string?"/>
-    <xsl:variable name="conref-topicid" as="xs:string">
-      <xsl:choose>
-        <xsl:when test="empty($topicid)">
-          <xsl:value-of select="/descendant-or-self::*[contains(@class, ' topic/topic ')][1]/@id"/>
-        </xsl:when>
-        <xsl:otherwise>
-          <xsl:value-of select="$topicid"/>
-        </xsl:otherwise>
-      </xsl:choose>
-    </xsl:variable>
-    <xsl:variable name="href-topicid" select="dita-ot:get-topic-id(.)" as="xs:string?"/>
-    <xsl:variable name="href-elemid" select="dita-ot:get-element-id(.)" as="xs:string?"/>
-    <xsl:variable name="conref-gen-id" as="xs:string">
-      <xsl:choose>
-        <xsl:when test="empty($elemid) or $elemid = $href-elemid">
-          <xsl:value-of select="generate-id((key('id', $conref-topicid)[contains(@class, ' topic/topic ')]//*[@id = $href-elemid])[1])"/>
-        </xsl:when>
-        <xsl:otherwise>
-          <xsl:value-of select="generate-id((key('id', $conref-topicid)[contains(@class, ' topic/topic ')]//*[@id = $elemid]//*[@id = $href-elemid])[1])"/>
-        </xsl:otherwise>
-      </xsl:choose>
-    </xsl:variable>
-    <xsl:variable name="href-gen-id" as="xs:string">
-      <xsl:variable name="topic" select="key('id', $href-topicid)"/>
-      <xsl:value-of select="generate-id($topic[contains(@class, ' topic/topic ')]//*[@id = $href-elemid][generate-id(ancestor::*[contains(@class, ' topic/topic ')][1]) = generate-id($topic)])"/>
-    </xsl:variable>
     <xsl:choose>
-      <xsl:when test="($conref-gen-id = '') or (not($conref-gen-id = $href-gen-id))">
-        <!--href target is not in conref target -->
-        <xsl:value-of select="$current-relative-path"/>
-        <xsl:value-of select="$conref-filename"/>
-        <xsl:value-of select="."/>
-      </xsl:when>
-      <xsl:when test="$conref-gen-id = $href-gen-id">
-        <xsl:text>#</xsl:text>
-        <xsl:value-of select="$conref-source-topicid"/>
-        <xsl:text>/</xsl:text>
-        <xsl:value-of select="$conref-gen-id"/>
-      </xsl:when>
+      <xsl:when test="exists($conref-filename)">
+        <xsl:attribute name="id" select="generate-id(..)"/>
+        <xsl:attribute name="conref:orig-id" select="."/>
```

Comment:
We generate a new ID for conreffed content as well as retain the original. We do this is step 1 so we don't have to generate any IDs in steps 2 or 3.